test/bd-chains-and-dma-tasks/dma-tasks-to-npu/good-5.mlir

```diff
@@ -0,0 +1,35 @@
+//
+// This file is licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+// (c) Copyright 2024 AMD Inc.
+
+// REQUIRES: ryzen_ai
+//
+// RUN: aie-opt --aie-assign-buffer-addresses --aie-dma-tasks-to-npu %s | FileCheck %s
+
+module {
+  aie.device(npu1_4col) {
+    %tile_0_0 = aie.tile(0, 0)
+    %tile_0_1 = aie.tile(0, 1)
+    %tile_0_2 = aie.tile(0, 2)
+    // CHECK: %{{.*}} = aie.buffer(%tile_0_1) {address = [[ADDR1:[0-9]+]] {{.*}}}
+    %buf0 = aie.buffer(%tile_0_1) : memref<32xi8> 
+    // CHECK: %{{.*}} = aie.buffer(%tile_0_1) {address = [[ADDR2:[0-9]+]] {{.*}}}
+    %buf1 = aie.buffer(%tile_0_1) : memref<32xi8> 
+
+    aiex.runtime_sequence(%arg0: memref<32xi8>) {
+      %t1 = aiex.dma_configure_task(%tile_0_0, MM2S, 0) {
```

Comment:
Should this be `%tile_0_1` ?

Reply:
Yes, this is wrong. Thanks for paying close attention. This should also generate an error I believe, unless we can access memory from mem tiles from the shim directly?

Reply:
[This](https://github.com/Xilinx/mlir-aie/blob/d23a3ab31b6dac9fc93be05a697f0ffafb0dcc00/lib/Dialect/AIE/IR/AIETargetModel.cpp#L353) allowed this to pass without error, specifically since `isMemNorth` evaluates to true for tile (0, 0) (I believe).

Since this is encoded in the target model ... can the shim tile in fact access mem tile memory directly?

Reply:
The shim tile cannot access memtile memory

Reply:
Please make a fix with a test if you can